### ISSUE TITLE
How to get the related class labels from Mosaic augmentation? 

### ISSUE THREAD

**Issue by innat** [ultralytics/yolov5#1144]
I'm trying to get the related class label from mosaic augmentation for multi-class object detection. I'm not using `Yolo5` but the mosaic implementation in my case inspired by Yolo implementation. I described my query in more detail over here, [SO](https://stackoverflow.com/questions/64335735/how-to-get-class-label-from-mosaic-augmentation-with-pytorch-dataloader). Please, have a look.

**Comment by glenn-jocher** [comment]
@innat if you are looking for multi-class object detection, I would simply use YOLOv5, which already incorporates all the mosaic augmentation you want.

**Comment by innat** [comment]
@glenn-jocher Could you please point me to the specific code snippet where the mosaic augmentation in Yolo5 deals with such cases? So, that I can understand when it randomly making a mosaic, how it tracks the labels in final results.

**Comment by glenn-jocher** [comment]
@innat of course. load_mosaic() in datasets.py is responsible for all mosaic activity. Your dataset trains with Mosaic by default in YOLOv5, no extra action is needed on your part, simply `python train.py --data custom.yaml`

https://github.com/ultralytics/yolov5/blob/4d3680c81dad760e5d1ffe397ab2fc9169d9fb70/utils/datasets.py#L643-L699

**Comment by glenn-jocher** [comment]
You can also view your mosaics during training. Suggest you start with the custom training tutorial, it explains all of this already.
https://docs.ultralytics.com/yolov5

![train_batch0](https://user-images.githubusercontent.com/26833433/96108045-6f221d00-0edd-11eb-84b8-c7fb2bb5e9d3.jpg)

**Comment by innat** [comment]
Thanks for the pointer and suggestion. -)

**Comment by innat** [comment]
The problem is solved. Here is the [answer](https://stackoverflow.com/a/64380170/9215780), leaving it here for future readers. 

![](https://i.stack.imgur.com/M1Ns7.png)
